.github/workflows/relay.yml

```diff
@@ -27,4 +27,4 @@ jobs:
       - name: Test
         run: go test ./pkg/... -v
       - name: Test with the race detector enabled
-        run: go test ./pkg/... -v -race -count=10
+        run: go test ./pkg/... -v -race -count=1 -timeout=15m
```

Comment:
The were primarily two reasons:
- 10 iterations take forever to complete now with the txm tests added
- There is an error in the tests: it fails starting from the second iteration mismatching the prom counters. My first idea was that tests do not cleanup properly after completion but I wasn't able to catch it immediately. Sure I can dig into it more, it's just not the first priority.

Reply:
Race detection is not meant to pass tests, and timeout parameter is how we bound the runtime

Reply:
I would recommend using shuffle like we do in core

Reply:
Those tests are failing with count > 1 without race flag too. To replicate on core develop branch:
`go test ./core/chains/solana/soltxm -v -count=3`

I'll have another look tomorrow, it could be I'm missing something.

Reply:
> There is an error in the tests: it fails starting from the second iteration mismatching the prom counters. My first idea was that tests do not cleanup properly after completion but I wasn't able to catch it immediately. Sure I can dig into it more, it's just not the first priority.

ah right that set of tests doesn't work well with race tests cuz of prom

another work around could be to tag that test file with `//go:build integration`

Reply:
on another note: we need have the CI also run the integration tests so the non-race tests should have `-tags integration` added to it